domain/account/account.go

```diff
@@ -59,20 +72,28 @@ func (auc *UseCase) GetBalance(id string) (int, error) {
 }
 
 //GetAccounts returns all API accounts
-func (auc *UseCase) GetAccounts() []domain.Account {
-	accounts := auc.StoredAccount.ReturnAccounts()
+func (auc *UseCase) GetAccounts() ([]domain.Account, error) {
+	accounts, err := auc.StoredAccount.ReturnAccounts()
+
+	if err != nil {
+		return []domain.Account{}, domain.ErrInsert
+	}
+
 	var account []domain.Account
 
 	for _, a := range accounts {
 		account = append(account, ChangeAccountStorage(a))
 	}
 
-	return account
+	return account, nil
 }
 
 // SearchAccount returns the account via the received ID
-func (auc UseCase) SearchAccount(id uuid.UUID) domain.Account {
-	accounts := auc.StoredAccount.ReturnAccounts()
+func (auc UseCase) SearchAccount(id uuid.UUID) (domain.Account, error) {
+	accounts, err := auc.StoredAccount.ReturnAccounts()
```

Comment:
since you're using an SQL db now, why not delegate this search by a given id to it?

by doing it you'll achieve a better performance and clean this seek code below.